tools/export/cdt/__init__.py

```diff
@@ -38,19 +38,22 @@ class Eclipse(Makefile):
     """Generic Eclipse project. Intended to be subclassed by classes that
     specify a type of Makefile.
     """
-    def get_target_config(self, ctx, configuration):
+    def get_target_config(self, configuration):
         """Retrieve info from cdt_definitions.json"""
-        tgt = deepcopy(TARGET_MAP[self.target])
-        defaults = deepcopy(_CONFIGS_OPTIONS['default'])
+        defaults = _CONFIGS_OPTIONS['default']
         eclipse_config = deepcopy(defaults['generic'])
         if configuration in defaults:
             eclipse_config.update(defaults[configuration])
 
+        # Get target-specific options. Use labels to find alias
+        # for target. This allows to define options within inheritance path
         target_specific = _CONFIGS_OPTIONS['targets']
-        if tgt.name in target_specific:
-            eclipse_config.update(target_specific[tgt.name]['generic'])
-            if configuration in target_specific[tgt.name]:
-                eclipse_config.update(target_specific[tgt.name][configuration])
+        aliases = self.toolchain.target.labels
+        for target_alias in aliases:
```

Comment:
Given that `aliases` is only ever used on the following line, I suggest combining the two into:
```python
for target_alias in self.toolchain.target.labels:
```